rust-src/concordium_base/src/id/sigma_protocols/com_ineq.rs

```diff
@@ -0,0 +1,174 @@
+//! The module provides the implementation of the sigma protocol for "proof of
+//! inequality for committed value and public value". This protocol enables one
+//! to prove that a committed value is not equal to a public one , without
+//! revealing the value.
+
+use super::{
+    com_mult::{ComMult, ComMultSecret, Witness as ComMultWitness},
+    common::{prove as sigma_prove, verify as sigma_verify, SigmaProof},
+};
+use crate::{
+    common::*,
+    curve_arithmetic::Curve,
+    pedersen_commitment::{Commitment, CommitmentKey, Randomness, Value},
+    random_oracle::RandomOracle,
+};
+use ff::Field;
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize)]
+pub struct Witness<C: Curve> {
+    // The witness consists of a com_mult witness and a commitment.
+    com_mult_witness: SigmaProof<ComMultWitness<C>>,
+    aux_com:          Commitment<C>,
+}
+
+/// Function for proving that a committed value `value` is different from
+///  `pub_value`. The parameters are
+/// - `com_key` - the commitment key used to commit to the value
+/// - `value` - the value inside the commitment
+/// - `value_tilde` - the randomness used to commit
+/// - `pub_value` - the public value claimed to be different from `value`
+/// - `csprng` - a cryptographically secure random number generator
+
+pub fn prove_com_ineq<R: rand::Rng, C: Curve>(
+    com_key: &CommitmentKey<C>,
+    value: &Value<C>,
+    value_tilde: &Randomness<C>,
+    pub_value: C::Scalar,
+    csprng: &mut R,
+) -> Option<Witness<C>> {
+    let mut transcript = RandomOracle::domain(b"InequalityProof");
+
+    let c = com_key.hide(&value, value_tilde);
+    transcript.append_message(b"g", &com_key.g);
+    transcript.append_message(b"h", &com_key.h);
```

Comment:
Why not just
```suggestion
    transcript.append_message(b"commitmentKey", &com_key);
```

Is there a reason for splitting them?

Reply:
Ok, sure. No specific reason for splitting.